changelogs/fragments/8909-flatpak-improve-name-parsing.yaml

```
@@ -0,0 +1,2 @@
+minor_changes:
+  - flatpak - improve `_parse_flatpak_name` function to better validation of flatpak application IDs according to official guidelines (https://github.com/ansible-collections/community.general/pull/8909).
```

I would make this more user-centric (also the product name Flatpak is capitalized):
```suggestion
  - flatpak - improve parsing of Flatpak names to improve validation of Flatpak application IDs according to official guidelines (https://github.com/ansible-collections/community.general/pull/8909).
```

---

I have simplified it slightly, but hopefully this will be helpful. You can check the changes.